common/src/main/java/tachyon/util/CommonUtils.java

```diff
@@ -157,25 +157,37 @@ public static ByteBuffer cloneByteBuffer(ByteBuffer buf) {
    * Join each element in paths in order, separated by {@code TachyonURI.SEPARATOR}.
    * <p>
    * For example, {@code concatPath("/myroot/", "dir", 1L, "filename")} returns
-   * {@code "/myroot/dir/1/filename"}
+   * {@code "/myroot/dir/1/filename"}, or
+   * {@code concatPath("tachyon://myroot", "dir", "filename")} returns
+   * {@code "tachyon://myroot/dir/filename"}. Note that a null element in paths
+   * is treated as empty string, i.e., "".
    *
    * @param paths to concatenate
    * @return joined path
+   * @throws IllegalArgumentException
    */
-  public static String concatPath(Object... paths) {
+  public static String concatPath(Object... paths) throws IllegalArgumentException {
     List<String> trimmedPathList = new ArrayList<String>();
-    for (int k = 0; k < paths.length; k ++) {
-      String path = paths[k].toString().trim();
-      String trimmedPath;
-      if (k == 0) {
-        trimmedPath = CharMatcher.is(TachyonURI.SEPARATOR.charAt(0)).trimTrailingFrom(path);
-      } else {
-        trimmedPath = CharMatcher.is(TachyonURI.SEPARATOR.charAt(0)).trimFrom(path);
-        if (trimmedPath == "") {
-          continue;
-        }
+    if (paths == null) {
+      throw new IllegalArgumentException("Can not concatenate a null set of paths.");
+    }
+    if (paths.length > 0 && paths[0] != null && !paths[0].toString().isEmpty()) {
+      trimmedPathList.add(CharMatcher.is(
+          TachyonURI.SEPARATOR.charAt(0)).trimTrailingFrom(paths[0].toString().trim()));
```

Comment:
Oh I was thinking to make a helper method that just does `CharMatcher.is(TachyonURI.SEPARATOR.charAt(0)).trimTrailingFrom(input.toString().trim())`
Where `input` is the string passed in. Then you can call the function something descriptive and when looking at the code logically the developer does not need to do the string manipulation in his/her head.